### ISSUE TITLE
user invites on homepage: ship to on-prem customers

### ISSUE THREAD

**Issue by slimsag** [sourcegraph/sourcegraph#33465]
We [shipped user invites on the homepage](https://github.com/sourcegraph/sourcegraph/issues/31214#issuecomment-1064508352) for all Sourcegraph.com Cloud users, with the intent of reviewing it's impact and collecting feedback for 2 weeks before shipping to on-prem.

Before we ship to on-prem, we should:

* [x] Verify the impact this change has had: For the entire month of March, just 52 signups came through this. https://analytics.amplitude.com/sourcegraph/chart/d4oba5a?source=workspace
* [ ] [exclude existing users based on name, not just email](https://github.com/sourcegraph/sourcegraph/issues/33463) (~1 week est.)
* [ ] Decide if we should continue investing in this.

My opinion: wait to see what signup % looks like in April, if ~10% then we should continue refining this. Otherwise, it is my belief that the current ~3.49% of signups it brings does not justify the real estate it takes up on the homepage and [concerns it may create about brand](https://sourcegraph.slack.com/archives/C01LZKLRF0C/p1647988171610719).

**Comment by muratsu** [comment]
Closing this issue as we've upcoming changes to [homepage](https://sourcegraph.slack.com/archives/CNC4YAL1E/p1649168585765889?thread_ts=1646773804.325119&cid=CNC4YAL1E) that would make this change less relevant.

cc: @slimsag , @a-bergevin , @rrhyne

**Comment by rrhyne** [comment]
@muratsu that is referring to the about site, not the app's authenticated home page.

@slimsag there was an error in that chart. We're actually at 4.5% for all invites. 

I can't tell who was invited from the home page panel and who was invited from the post sign-up panel. I'm not sure if that's possible without changing SignUpInvitedByUser to two events: SignUpInvitedByUserPostInstall and SignUpInvitedByUserHomepage. WDYT?

**Comment by slimsag** [comment]
> We're actually at 4.5% for all invites.

👍 

> I can't tell who was invited from the home page panel and who was invited from the post sign-up panel.

Yup, you're right, we don't know this at present and would have to split the events by page I think as you said.

**Comment by muratsu** [comment]
Reopening this as I mixed the about homepage with the product homepage

**Comment by a-bergevin** [comment]
@slimsag how much work would it be to split the events by page as Rob mentions above? I think we could let this continue to run for several weeks before making a decision, but I think we want clearer data on where exactly those invites are coming from (by page).

**Comment by slimsag** [comment]
Not too hard, maybe an hour of work. @philipp-spiess may be able to chime in here too.

The event is logged here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@9479bd249a46c838e0734c863e43ad4ce82cebc1/-/blob/client/web/src/auth/SignUpPage.tsx?L48-54

I think we'd just need to change that to take into account another parameter, right now it's like `/sign-up?invitedBy=user` and we'd need `/sign-up?invitedBy=user&from=home` and `/sign-up?invitedBy=user&from=postinstall` to map to those events.

**Comment by a-bergevin** [comment]
Thanks @slimsag , also that's the first time I've seen someone use that code linking from our product and man that's convenient. :)

@brady-garrett do we have naming conventions/standards around how we handle events to make the analytics simpler? Any thoughts here?

**Comment by philipp-spiess** [comment]
> I think we'd just need to change that to take into account another parameter, right now it's like `/sign-up?invitedBy=user` and we'd need `/sign-up?invitedBy=user&from=home` and `/sign-up?invitedBy=user&from=postinstall` to map to those events.

Yes, changing the URL is likely the meat of this change. It will also take a bit until all of the old invite URLs disappear in peoples inboxes and we start to get meaningful data I think. Let me know if we want to go ahead with this change and we can to ship the changes asap.
